import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x =sc.nextInt();
        
        int y =sc.nextInt();
        int z = sc.nextInt();
        
        int a [] = new int[x];
        int b [] = new int[y];
        int c [] = new int[z];
        
        for(int i = 0 ; i < x ; i++){
            a[i] = sc.nextInt();
        }
        for(int i = 0 ; i < y ; i++){
            b[i] = sc.nextInt();
        }
        for(int i = 0 ; i < z ; i++){
            c[i] = sc.nextInt();
        }
        
        for(int i = 0 ; i< x ; i++){
            for(int j = 0 ; j< y ; j++){
                for(int k = 0 ; k < z ; k++){
                    if(a[i] == b[j] && b[j] == c[k]){
                        System.out.print(a[i]+" ");
                    }
                }
            }
        }
        
        }
}